app/src/main/java/com/appunite/loudius/analytics/EventParameter.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2023 AppUnite S.A.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.appunite.loudius.analytics
+
+sealed class EventParameter<T> {
```

Comment:
**opinion**: when using sealed class, it is usually a little bit simpler to just use inner classes, especially if all are specified in a single file:

```kotlin
sealed class EventParameter<T> {
   abstract val name: String
   abstract val value: T
   data class String(
    override val name: String,
    override val value: String
) : EventParameter<String>()
data class Boolean(
    override val name: String,
    override val value: Boolean
) : EventParameter<Boolean>()
```

* This gives better autocompletion: If you Type `EventParameter.` than IDE will help you fill the rest, and present all possible autocomplitions.
* It gives you standard naming for sealed class. In this case you're naming class like: `StringEventParameter`, but it could be also: `EventParameterString`, or also `ParameterString`. In case of inner sealed classes you always have: `EventParameter.<Somting>`

**suggestion**:
You may consider skipping using generics, because `value` doesn't need to be in the base class. I mean something like this :
```kotlin
sealed class EventParameter {
    abstract val name: String
}

data class StringEventParameter(
    override val name: String,
    val value: String
) : EventParameter()

data class BooleanEventParameter(
    override val name: String,
    val value: Boolean
) : EventParameter()
```

You won't need to type `EventParameter<*>` everywhere :)

It's a little simple, but both solutions are correct :) 

All together:
```kotlin
sealed class EventParameter {
    abstract val name: String

    data class String(override val name: String, val value: String) : EventParameter()

    data class Boolean(override val name: String, val value: Boolean) : EventParameter()
}
```

Reply:
How simple!🤩